app/views/spree/admin/products/new.html.haml

```diff
@@ -0,0 +1,117 @@
+          = f.field_container :unit_name do
+            = f.label :product_variant_unit_name, t(:unit_name)
+            %input.fullwidth{ id: 'product_variant_unit_name','ng-model' => 'product.variant_unit_name', :name => 'product[variant_unit_name]', :placeholder => t('admin.products.unit_name_placeholder'), :type => 'text' }
+      .twelve.columns.alpha
+        .six.columns.alpha
+          = render 'spree/admin/products/primary_taxon_form', f: f
+        .two.columns
+          = f.field_container :price do
+            = f.label :price, t(:price)
+            %span.required *
+            %br/
+            = f.text_field :price, class: 'fullwidth'
+            = f.error_message_on :price
+        .two.columns
+          = f.field_container :on_hand do
+            = f.label :on_hand, t(:on_hand)
+            %span.required *
+            %br/
+            = f.text_field :on_hand, class: 'fullwidth'
+            = f.error_message_on :on_hand
+        .two.columns.omega
+          = f.field_container :on_demand do
+            = f.label :on_demand, t(:on_demand)
+            %br/
+            = f.check_box :on_demand
+            = f.error_message_on :on_demand
+      .twelve.columns.alpha
+        .six.columns.alpha &nbsp;
+        .three.columns
+          - if Spree::TaxCategory.any?
+            = render 'spree/admin/products/tax_category_form', f: f
+          - else
+            &nbsp;
+        .three.columns.omega
+          = render 'spree/admin/products/shipping_category_form', f: f
+
+      .twelve.columns.alpha
+        = f.field_container :description do
+          = f.label :product_description, t(:product_description)
+          %br/
+          %text-angular{'id' => 'product_description', 'name' => 'product[description]', 'class' => 'text-angular', 'textangular-strip' => true, 'ta-paste' => "stripFormatting($html)", 'ta-toolbar' => "[['bold','italics','clear']]"}
+          = f.error_message_on :description
+  .four.columns.omega{ style: "text-align: center" }
+    %fieldset.no-border-bottom{ id: "image" }
+      %legend{align: "center"}= t(:image)
+      .row
+        = image_tag "noimage/product.png", class: "four columns alpha"
+      .row
+        = f.fields_for 'images_attributes[]', f.object.images.build do |image_fields|
+          = image_fields.file_field :attachment
+  .sixteen.columns.alpha
+    .form-buttons.filter-actions.actions{ 'data-hook' => "buttons" }
+      = button t('actions.create'), 'icon-ok', :submit, value: "create"
+      %span.or
+        = t(:or)
+      = button t('actions.create_and_add_another'), 'icon-repeat', :submit, value: 'add_another'
+      %span.or
+        = t(:or)
+      = link_to_with_icon 'icon-remove', t('actions.cancel'), admin_products_path, :class => 'button'
+
+
+:javascript
+  angular.element(document.getElementById("new_product")).ready(function() {
+    angular.bootstrap(document.getElementById("new_product"), ['admin.products']);
+  });
```

Comment:
@sauloperez what should we do about this ?

Reply:
I guess this the way to "boot" Angular, so I'd keep it. Have we solved it differently in other cases?

Reply:
this is manual bootstrapping.
https://docs.angularjs.org/api/ng/function/angular.bootstrap
it's the same as a ng-app tag.
https://stackoverflow.com/questions/35018399/ng-app-or-angular-bootstrap

I think we can replace this with ng-app on the new_product element.
like
%fieldset.no-border-bottom{ id: "new_product", ng-app: "admin.products" }
we may need to define the correct controller in the same line like here:
app/overrides/spree/admin/products/_form/add_units_form.html.haml.deface:
    3: .variant_units_form{ 'ng-app' => 'admin.products', 'ng-controller' => 'editUnitsCtrl' }

Reply:
can you give it a try Hugo and see if it works?
otherwise, we keep it as is...

Reply:
Applying ng-app to an element is indeed the solution. We can also use something like 
``` - content_for :app_wrapper_attrs do
		  = "ng-app='admin.products'"
	```
Which will instantiate angular on the whole page, in case we need to use it outside of that one element.

Reply:
I tried content_for at the top of the file and ng-app on the element, but it doesn't seem to work. This module is used to display the text area for description, with the two solutions I mention here it's present in the DOM but doesn't render. Could the `angular.element(document.getElementById("new_product")).ready(function()` have something to do with it ?